Test/ghost/PhantomData.dfy

```diff
@@ -0,0 +1,63 @@
+// RUN: %dafny /compile:0 /dprint:- "%s" /env:0 > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype PhantomData<V> = PhantomData(ghost data: V)
+
+method ParsingGhostActualBindings()
+{
+    // Check that the TupleTypeName of the following tuples is the same
+    var tuple0 := (ghost 10:=10, 3:=3, 0:=0, 1:=1, 2:=2, 4:=4, 5:=5, 6:=6, 7:=7, 8:=8, 9:=9);
+    var tuple1 := (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, ghost 10);
+    assert tuple0 == tuple1;
+}
+
+method GhostBad1()
+{
+    var x := (ghost 123); // ghost not allowed in parenthesized expression
+}
+
+method GhostBad2()
+{
+    ghost var x: (ghost int) := 123; // ghost not allowed in parenthesized type
+}
+
+method GhostBad3()
+{
+    var (ghost x) := 123; // TODO(fpoli): Is it fine to disallow this?
+}
+
+method GhostToNonGhost1(ghost a: int) returns (q: int)
+{
+    var x1 := a; // error: ghost to non-ghost
+    ghost var x2 := a;
+    q := a; // error: ghost to non-ghost
+}
+
+method GhostToNonGhost2(b: (int, ghost (int, bool), int))
+{
+    var y1 := b.0;
+    var y2 := b.1.0; // error: ghost to non-ghost
+    var y3 := b.1.1; // error: ghost to non-ghost
+    ghost var y4 := b.0;
+    ghost var y5 := b.1.0;
+    ghost var y6 := b.1.1;
+    var (y7, ghost (y8, y9), y10) := b;
+    var y11 := y10; // Check that y11 has not been parsed as ghost
+    var (ghost y12, ghost y13) := b.1;
+}
+
+// TODO(fpoli): The following signature causes the error "Undeclared top-level type or type parameter: _tuple#2OG". What does it mean?
```

Comment:
Seems like a runtime artefact. Maybe it goes away if we change the TupleTypeName above to `$"_tuple#{NonGhostDims}"`

Reply:
I'll try, thanks. I didn't expect dafny with `/compile:0` to generate runtime stuff.

Reply:
It seems to me that `theBuiltIns.TupleType` has not been called for `new bool[2] { false, true }`.

Reply:
I fixed the bug.